sycl::pown fails on integrated Intel GPU

**Bug summary**
A simple kernel with `sycl::pown` doesn't JIT compile on an integrated Intel GPU (in Intel's DevCloud). Instead it fails with a runtime error:
```
[hipSYCL Error] from /home/u183646/hipsycl/source/src/runtime/ze/ze_code_object.cpp:127 @ ze_executable_object(): ze_executable_object: Couldn't create module handle
Build log: 
error: undefined reference to `__builtin_spirv_OpenCL_pown_f64_i64'
in function: '__builtin_spirv_OpenCL_pown_f64_i64' called by kernel: 'void __hipsycl_sscp_kernel<hipsycl::glue::__sscp_dispatch::single_task<main::$_0> >(hipsycl::glue::__sscp_dispatch::single_task<main::$_0> const&)'

error: backend compiler failed build.
 (error code = ze:1879048196)
[hipSYCL Error] from /home/u183646/hipsycl/source/src/runtime/ze/ze_code_object.cpp:127 @ ze_executable_object(): ze_executable_object: Couldn't create module handle
Build log: 
error: undefined reference to `__builtin_spirv_OpenCL_pown_f64_i64'
in function: '__builtin_spirv_OpenCL_pown_f64_i64' called by kernel: 'void __hipsycl_sscp_kernel<hipsycl::glue::__sscp_dispatch::single_task<main::$_0> >(hipsycl::glue::__sscp_dispatch::single_task<main::$_0> const&)'

error: backend compiler failed build.
 (error code = ze:1879048196)
[hipSYCL Error] from /home/u183646/hipsycl/source/include/hipSYCL/runtime/kernel_cache.hpp:421 @ get_or_construct_code_object_impl(): kernel_cache: code object creation has failed
[hipSYCL Error] from /home/u183646/hipsycl/source/src/runtime/ze/ze_queue.cpp:611 @ submit_sscp_kernel_from_code_object(): ze_queue: Code object construction failed
0
============== hipSYCL error report ============== 
hipSYCL has caught the following undhandled asynchronous errors: 

   0. from /home/u183646/hipsycl/source/src/runtime/ze/ze_code_object.cpp:127 @ ze_executable_object(): ze_executable_object: Couldn't create module handle
Build log: 
error: undefined reference to `__builtin_spirv_OpenCL_pown_f64_i64'
in function: '__builtin_spirv_OpenCL_pown_f64_i64' called by kernel: 'void __hipsycl_sscp_kernel<hipsycl::glue::__sscp_dispatch::single_task<main::$_0> >(hipsycl::glue::__sscp_dispatch::single_task<main::$_0> const&)'

error: backend compiler failed build.

   1. from /home/u183646/hipsycl/source/src/runtime/ze/ze_code_object.cpp:127 @ ze_executable_object(): ze_executable_object: Couldn't create module handle
Build log: 
error: undefined reference to `__builtin_spirv_OpenCL_pown_f64_i64'
in function: '__builtin_spirv_OpenCL_pown_f64_i64' called by kernel: 'void __hipsycl_sscp_kernel<hipsycl::glue::__sscp_dispatch::single_task<main::$_0> >(hipsycl::glue::__sscp_dispatch::single_task<main::$_0> const&)'

error: backend compiler failed build.

   2. from /home/u183646/hipsycl/source/include/hipSYCL/runtime/kernel_cache.hpp:421 @ get_or_construct_code_object_impl(): kernel_cache: code object creation has failed
   3. from /home/u183646/hipsycl/source/src/runtime/ze/ze_queue.cpp:611 @ submit_sscp_kernel_from_code_object(): ze_queue: Code object construction failed
The application will now be terminated.
terminate called without an active exception
Aborted
``` 

**To Reproduce**
Simply compile the following code snippet:
```c++
#include <iostream>
#include "sycl/sycl.hpp"

int main() {
  sycl::queue q{};
  double* val = sycl::malloc_shared<double>(1, q);
  
  q.single_task([=]() {
      *val = sycl::pown(3.1415, 2);
  }).wait();
  
  std::cout << *val << std::endl;
  return 0;
}
```
with `syclcc -std=c++17 -sycl-std=2020 --opensycl-target="generic" main.cpp`.


**Expected behavior**
It should compile and run, as it does with hipSYCL on NVIDIA or AMD GPUs as well as with DPC++ on the Intel GPU.

**Describe your setup**
The hipSYCL commit `caee556` has been compiled with
```
cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=../caee5569 -DWITH_ACCELERATED_CPU=ON -DCMAKE_CXX_COMPILER=clang++ -DWITH_SSCP_COMPILER=ON -DWITH_LEVEL_ZERO_BACKEND=ON ../source
```
where `clang++` has been installed via `spack install llvm +clang -gold` (clang 16.0.6).


**Optional additional diagnostic information**
- `HIPSYCL_DRYRUN=1`:  [HIPSYCL_DRYRUN.txt](https://github.com/OpenSYCL/OpenSYCL/files/12142537/HIPSYCL_DRYRUN.txt)
- `HIPSYCL_DEBUG_LEVEL=3`:  [HIPSYCL_DEBUG_LEVEL_3.txt](https://github.com/OpenSYCL/OpenSYCL/files/12142507/HIPSYCL_DEBUG_LEVEL_3.txt)
- `syclcc --hipsycl-version`: [HIPSYCL_VERSION.txt](https://github.com/OpenSYCL/OpenSYCL/files/12142545/HIPSYCL_VERSION.txt)
- `opensycl-info`: [OPENSYCL_INFO.txt](https://github.com/OpenSYCL/OpenSYCL/files/12142526/OPENSYCL_INFO.txt)

--- a/src/libkernel/sscp/spirv/math.cpp
+++ b/src/libkernel/sscp/spirv/math.cpp
@@ -163,10 +163,11 @@ HIPSYCL_SSCP_MAP_BUILTIN_TO_SPIRV_BUILTIN2(nextafter)
 HIPSYCL_SSCP_MAP_BUILTIN_TO_SPIRV_BUILTIN2(powr)
 
 float __spirv_ocl_pown(float, __hipsycl_int32);
-double __spirv_ocl_pown(double, __hipsycl_int64);
+double __spirv_ocl_pown(double, __hipsycl_int32);
 
 HIPSYCL_SSCP_BUILTIN float __hipsycl_sscp_pown_f32(float x, __hipsycl_int32 y) { return __spirv_ocl_pown(x, y); }
-HIPSYCL_SSCP_BUILTIN double __hipsycl_sscp_pown_f64(double x, __hipsycl_int64 y) {return __spirv_ocl_pown(x, y); }
+HIPSYCL_SSCP_BUILTIN double __hipsycl_sscp_pown_f64(double x, __hipsycl_int64 y)
+{return __spirv_ocl_pown(x, static_cast<__hipsycl_int32>(y)); }
 
 HIPSYCL_SSCP_MAP_BUILTIN_TO_SPIRV_BUILTIN2(remainder)
 HIPSYCL_SSCP_MAP_BUILTIN_TO_SPIRV_BUILTIN(rint)

